src/config/plugin.ts

```diff
@@ -56,6 +56,33 @@ function processCommandIds(files: string[]): string[] {
   })
 }
 
+function determineCommandDiscoveryOptions(
+  commandDiscovery: string | CommandDiscovery | undefined,
+): CommandDiscovery | undefined {
+  if (!commandDiscovery) return
```

Comment:
to simplify types and undefined checks further down, what if it always returned a CommandDiscovery (and the return from `undefined` sets it to the default behavior?

Reply:
There are instances where there's not any command discovery, i.e. plugins that only have hooks but no commands (e.g. plugin-not-found)

Setting to the default would probably be fine, since it just wouldn't find any commands but then you're incurring the cost of searching for commands when we already know that none will be found